resources/acsl/datastructures.md

```diff
@@ -40,32 +56,169 @@
 
 ## Terminology
 
+Trees consist of elements called *nodes* that are connected by *edges*. The *root* is the top node in the tree. *Parent* nodes are nodes that have other nodes, *children* nodes, stemming from them. *Leaves* are the nodes that do not have any other children nodes stemming from them. Nodes that have the same immediate parent are *siblings*.
+
+So, in the following table below, 7 is the root. 7, 2, and 9 are all parent nodes. 2, 9, 1, 5, and 14 are all children nodes; notice that parent nodes can also be children nodes of other nodes. 1, 5, and 14 are leaves. 1 and 5 are siblings.
+
+<img src="https://miro.medium.com/proxy/1*4M5MU3CqJYGNExEi5Ttuew.png" width="150" height="100" />
+
 ## Binary Search Trees
 
+Binary search trees are one way to store items in a particular order. They can efficiently insert, delete, and look for items within the tree. 
+
+Each node can have a total of two children. The left child must be less than or equal in value, whereas the right child must be greater in value. With numbers, this is easy enough; as for alphabet letters, they are considered to be "less than" another letter if they come earlier in the alphabet. So, for example, A < E.
+
 ### Inserting Nodes
 
+Inserting a node requires knowing its position relative to existing nodes. Here's an example with the word *AMERICAN*:
+
+| Step <img width=400/> | Description <img width=125/> | Step <img width=400/>| Description <img width=125/>|
+| --- | --- | --- | --- |
+| 1. <img src="https://user-images.githubusercontent.com/60682642/88571933-0032d780-d004-11ea-80ec-62f278e342a1.png" width="75" height="75" /> | *AMERICAN* starts with an *A*; thus, it is natural to make the tree's root an *A*. | 5. <img src="https://user-images.githubusercontent.com/60682642/88572623-11301880-d005-11ea-9df9-2f47d33f1063.png" width="150" height="150" /> | *I* belongs on the right of *A*. It belongs to the left of *M* as well. *I* comes later in the alphabet than *E*, so *I* is placed to the right of *E*. |
```

Comment:
I'll chuck these photos into the `res` folder.

Reply:
In the future, instead of uploading to GitHub or something, just give me the images and I'll throw them into `res` (link them directly there).

Also, I believe there's a nicer syntax for Markdown images (it's like ![](link) or something). Try to find what that is and use that.

Reply:
Or instead of giving me the images, I can show you how to edit the repository on your computer and you can do it yourself (This is the best solution, because it ensures you can see what the website will look like afterwards.)

Reply:
About the nicer syntax, I could only find **![image](link)** , but this didn't let me change the size of the picture, which is why I had to resort to the <img src=> thing.

Reply:
Also, editing the repository sounds good! I have no idea how many pictures I'll use in the future (especially when I get to digital electronics oh no), so I'd feel bad for having to ask you to put photos in the res folder all the time.